### PR TITLE
Add PyPi publishing on release and ergonomics changes

### PR DESCRIPTION
**Pull request by rdettai**
We would want to add back the automated release of VAST Python with the new implementation.

### CODE REVIEW

**Comment by rdettai** [comment]
Part of https://github.com/orgs/tenzir/projects/8/views/1?pane=issue&itemId=9361781

File: .github/workflows/vast.yaml [hunk 1]
```diff
@@ -1167,21 +1167,38 @@ jobs:
         run: |
           poetry config repositories.test-pypi https://test.pypi.org/legacy/
           poetry config pypi-token.test-pypi "${{ secrets.TEST_PYPI_TOKEN }}"
+      - name: Publish ${{ needs.configure.outputs.release-version }} to Test PyPI
+        working-directory: python
+        run: |
+          poetry publish --build -r test-pypi --no-interaction
```
Comment (dominiklohmann) [inline .github/workflows/vast.yaml:1173]: Instead of adding one additional job for this, can we just add these two steps to the `python` job right above?

**Review by dominiklohmann — APPROVED** [review]
Generally LGTM, mostly two smaller design questions.

File: .github/workflows/vast.yaml (hunk 1 above)
Comment (rdettai) [reply .github/workflows/vast.yaml:1173]: good point, we could actually also merge the release job, couldn't we?

File: python/pyproject.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [tool.poetry]
-name = "vast"
+name = "pyvast"
```
[the remark this answers is not in the document]
Comment (rdettai) [reply python/pyproject.toml:2]: Not sure about the pros and cons. Top level you mean `python/` or `python/vast/` ? For users they would use
```
pip install pyvast
```
and
```
from pyvast import xxx
```
What does it bring for the directory to be called `vast`? the way we call it internally.

File: .github/workflows/vast.yaml (hunk 1 above)
Comment (dominiklohmann) [reply .github/workflows/vast.yaml:1173]: I think that's cleaner to have separate given the on release condition, just to make sure there's no lingering debug build config that we package.

File: python/pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply python/pyproject.toml:2]: I meant moving `/python/vast/` into `/pyvast`, essentially back to where we started.

[the remark this answers is not in the document]
Comment (dominiklohmann) [reply python/pyproject.toml:2]: Okay then, I don't have too strong of an opinion on this.

[the remark this answers is not in the document]
Comment (rdettai) [reply python/pyproject.toml:2]: Actually, I looked into it and the `python/vast` dir should actually be named `python/pyvast` (see https://python-poetry.org/docs/basic-usage/):
>> Poetry assumes your package contains a package with the same name as tool.poetry.name located in the root of your project.

So even though that was not what you meant at the beginning, you are actually right that some renaming is required 😃

**Comment by rdettai** [comment]
The merge of #2869 was unintentional, but we are not going through the process of reverting. We'll proceed with the merge of both to master at once.

**Pull request merged**